### PR TITLE
Add TLS support for TCP transport

### PR DESCRIPTION
**Pull request by bernd**
Adding TLS support for TCP transports. Please review, thanks! :smiley:

### CODE REVIEW

File: src/main/java/org/graylog2/gelfclient/GelfConfiguration.java [hunk 1]
```diff
@@ -147,6 +151,66 @@ public GelfConfiguration queueSize(final int size) {
     }
 
     /**
+     * Check if TLS option for the transport is enabled.
+     *
+     * @return true if TLS is enabled, false if disabled
+     */
+    public boolean isTls() {
```
Comment (joschi) [inline src/main/java/org/graylog2/gelfclient/GelfConfiguration.java]: For clarity I would rename that method to `isTlsEnabled` or `tlsEnabled`. It doesn't follow the Java Beans spec, but this isn't exactly a Java Bean, is it?

File: src/main/java/org/graylog2/gelfclient/GelfConfiguration.java [hunk 2]
```diff
@@ -147,6 +151,66 @@ public GelfConfiguration queueSize(final int size) {
...
+        return tls;
+    }
+
+    /**
+     * Enable/disable TLS for transport.
+     *
+     * @param tls {@code true} to enable TLS, {@code false} to disable it
+     * @return {@code this} instance
+     */
+    public GelfConfiguration tls(final boolean tls) {
```
Comment (joschi) [inline src/main/java/org/graylog2/gelfclient/GelfConfiguration.java]: Maybe methods like `enableTls` and `disableTls` would fit better in the builder-like usage of this class.

File: src/main/java/org/graylog2/gelfclient/GelfConfiguration.java [hunk 3]
```diff
@@ -147,6 +151,66 @@ public GelfConfiguration queueSize(final int size) {
...
+        return tlsVerifyCert;
+    }
+
+    /**
+     * Enable/disable TLS certificate verification for transport.
+     *
+     * @param tlsVerifyCert {@code true} to enable TLS certificate verification, {@code false} otherwise
+     * @return {@code this} instance
+     */
+    public GelfConfiguration tlsVerifyCert(final boolean tlsVerifyCert) {
```
Comment (joschi) [inline src/main/java/org/graylog2/gelfclient/GelfConfiguration.java]: Maybe a method pair like `enableTlsVerification()` and `disableTlsVerification()` would fit better in the builder-like usage of this class.

File: src/main/java/org/graylog2/gelfclient/GelfConfiguration.java (hunk 1 above)
Comment (joschi) [inline src/main/java/org/graylog2/gelfclient/GelfConfiguration.java]: Although I'm well aware of the naming scheme of methods in this class. Maybe we want to update them as well in the future.

**Comment by joschi** [comment]
LGTM (despite the naming of methods in the configuration class) :+1:

Feel free to merge.

File: src/main/java/org/graylog2/gelfclient/GelfConfiguration.java (hunk 3 above)
Comment (bernd) [inline src/main/java/org/graylog2/gelfclient/GelfConfiguration.java]: Agreed, I will change that. Thanks!

**Pull request merged**